### ISSUE TITLE
Error: 'NoneType' object has no attribute 'NoWcsKeywordsFoundError'

### ISSUE THREAD

**Issue by fergalm** [astropy/astropy#3572]
My code is raising an attribute error that I don't believe should happen.

```
/home/fmullall/all/kepler/twowheel/whereToPoint/optimalField/optimalField.py in doPlotFc(img, hdr, ra_deg, dec_deg, cat)
    339     #Draw 35 pix circle
    340     radius_deg = 70/3600.
--> 341     w = wcs.WCS(hdr)
    342
    343

/home/fmullall/.local/lib64/python2.7/site-packages/astropy-0.2.3-py2.7-linux-x86_64.egg/astropy/wcs/wcs.pyc in __init__(self, header, fobj, key, minerr, relax, naxis, keysel, colsel, fix)
    312                                      relax=relax, keysel=keysel_flags,
    313                                      colsel=colsel)
--> 314             except _wcs.NoWcsKeywordsFoundError:
    315                 # The header may have SIP or distortions, but no core
    316                 # WCS.  That isn't an error -- we want a "default"

AttributeError: 'NoneType' object has no attribute 'NoWcsKeywordsFoundError'
```

Looking at the code, I suspect the problem starts near the top of wcs.py
```
try:
    from . import _wcs
except ImportError:
    _wcs = None
```

This problem, in turn, is probably due to my LD_LIBRARY_PATH not being set right, and python is unable to find _wcs.so. But if astropy can't find _wcs.so, shouldn't the error occur at that point, instead of much later when I try to use the code?

**Comment by mdboom** [comment]
`_wcs.so` is a Python extension module, so `LD_LIBRARY_PATH` doesn't come into play.

The reason it doesn't complain at import time is that the entire astropy tree is intended to be importable even if the C extensions have not been built.

So it seems that for some reason this extension isn't getting built and then found.  How did you build and install astropy?

**Comment by fergalm** [comment]
I'm using Anaconda. There's something particular about this machine because astropy and wcs have worked before for me on different boxes using Anaconda as the installer.

_wcs.so did compile, as I can see it in the astropy directory:
```
ls /home/fmullall/.local/lib64/python2.7/site-packages/astropy-0.2.3-py2.7-linux-x86_64.egg/astropy/wcs
docstrings.py*
docstrings.pyc
_docutil.py*
_docutil.pyc
include/
__init__.py*
__init__.pyc
setup_package.py*
setup_package.pyc
tests/
_wcs.py*
wcs.py*
_wcs.pyc
wcs.pyc
_wcs.so*
```

**Comment by astrofrog** [comment]
@fergalm - this is just a side issue, but if you are using anaconda you don't need to compile astropy. It looks like you have a very old version (0.2.3) installed in ``.local`` but you should remove that because anaconda can install the latest version of astropy using ``conda install astropy``.

**Comment by embray** [comment]
A warning in this case might be nice (the warning should be disabled if `_ASTROPY_SETUP_` is defined though).

**Comment by fergalm** [comment]
Ah, I think I see my problem. I recently changed machine, and I copied a lot of directories over blindly (including .local). Deleting the old version, compiled with an older glibc fixes my immediate problem. I'm now using astropy ver 0.4.2, and everything is working. Thanks.

Agreeing with @embray, I still think astropy should, at the very least, warn me when the core functionality of the Wcs class can't be loaded. That would have saved me quite a bit of debugging time.
